pylint/checkers/classes/class_checker.py

```diff
@@ -2059,6 +2059,11 @@ def _check_init(self, node: nodes.FunctionDef, klass_node: nodes.ClassDef) -> No
                 except astroid.InferenceError:
                     continue
 
+            # Skip if klass is Enum, Python's own docs and examples
+            # do not recommend Enum subclasses call Enum.__init__
+            if klass.name == "Enum":
```

Comment:
I think elsewhere we check the qname to avoid getting red herrings from other modules.
```suggestion
            if klass.qname() == "enum.Enum":
```